onnxruntime/core/graph/graph_proto_serializer.cc

```diff
@@ -43,12 +43,29 @@
   }
 
   if (include_initializer) {
+    std::unordered_set<std::string> current_scope_initializer_set;
+
     auto& initializers = graph_view.GetAllInitializedTensors();
     for (auto& it : initializers) {
       auto* p_initializer = graph_proto.add_initializer();
       *p_initializer = *(it.second);
+      current_scope_initializer_set.insert(it.first);
+    }
+
+    // handle outer scope value which is a constant initializer
+    for (auto& node_idx : graph_view.GetNodesInTopologicalOrder()) {
+      const auto& node = graph_view.GetNode(node_idx);
```

Comment:
Is the requirement to include outer scope values this way specific to TRT? 

e.g. if I have an EP that will use the GraphProto in a scenario where the outer scope values are still accessible, copying them into the GraphProto isn't necessary.

If this isn't a general requirement it may need to be optional.

Reply:
Good point, I will add an include_outer_scope_values argument for GraphViewerToProto() to make it optional.